lib/matplotlib/quiver.py

```diff
@@ -490,13 +502,25 @@ def __init__(self, ax, *args,
         self.headwidth = headwidth
         self.headlength = float(headlength)
         self.headaxislength = headaxislength
+        self.mid_scale = mid_scale
         self.minshaft = minshaft
         self.minlength = minlength
         self.units = units
         self.scale_units = scale_units
         self.angles = angles
         self.width = width
 
+        # Checks the boundaries of mid_scale if outside range default of 0.5
+        if 0.0 < mid_scale < 1.0:
+            self.mid_scale = mid_scale
+        else:
+            self.mid_scale = 0.5
```

Comment:
please error out in that case ("Errors should never pass silently.")

Reply:
I was thinking about this perhaps they should be left to be as the user specified and throw instead a warning that the arrows will be off the shaft?
If I have more time I would also think about allowing this to be a 1D array to allow the arrowhead position on the shaft to convey extra information. This would be a reason for mid_scale and head_pos to stay independent I think as your other comment.

Reply:
For now, I would just error out. We can always expand the functionality later on if there is demand.

Reply:
I have made the edits and committed changes